spec/lang/well-formed.md

```diff
@@ -382,13 +384,16 @@ impl ValueExpr {
                         }
                         Type::Int(IntType::usize_ty::<T>())
                     }
-                    VTableMethodLookup(_method) => {
-                        let Type::Ptr(PtrType::VTablePtr) = operand else {
+                    VTableMethodLookup(method) => {
+                        let Type::Ptr(PtrType::VTablePtr(trait_name)) = operand else {
                             throw_ill_formed!("UnOp::VTableMethodLookup: invalid operand: not a vtable pointer");
                         };
 
-                        // We do not statically have enough information to check that the method name exists in the right vtable,
-                        // but this is only a type safety problem, which we don't catch anyways.
+                        let Some(trait_methods) = prog.traits.get(trait_name) else {
+                            throw_ill_formed!("UnOp::VTableMethodLookup: invalid operand: trait of the vtable pointer doesn't exist");
```

Comment:
Is this even reachable? I would expect `operand` itself to not be well-formed rather than having a nonsense type.

Reply:
It was reachable because `PtrType::check_wf` and thus `Type::check_wf` didn't take the program as an argument and thus couldn't check this. Fixed now.